File: py/32.py
# 45228
import euler
p = {}
for a in xrange(1, 10000):
    for b in xrange(a+1, 10000 / a + 1):
        c = a * b
        aa = euler.digit_usage(a)
        bb = euler.digit_usage(b)
        cc = euler.digit_usage(c)
        if aa & bb & cc == 0 and aa | bb | cc == int('1111111110', 2):
            p[c] = True
print sum(p.keys())

File: hs/32.hs
-- 45228
import Data.List(nub)
import Euler(digitUsage, fromDigits, intSqrt)

kk = 9

isPandigital n a b = digitUsage n == xs
    where xs = zipWith3 (\x y z -> x+y+z)
                 (digitUsage a) (digitUsage b) (digitUsage $ a*b)

genPandigital m n = [x*y | x <- [1..intSqrt m],
                           y <- [x+1..m `div` x], isPandigital n x y]

sumPandigital k = sum $ nub $ genPandigital (fromDigits ys) (fromDigits xs)
    where xs = reverse [1..k]
          n = (k-1) `div` 2
          ys = take n xs

main = putStrLn $ show $ sumPandigital kk
